### PR TITLE
Suggest Jenkins Configuration as Code plugin in the installation wizard

### PR DESCRIPTION
**Pull request by jetersen**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Proposed changelog entries

* Suggest Jenkins Configuration as Code plugin in the installation wizard
* ...

### Submitter checklist

- [ ] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [x] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@oleg-nenashev

### CODE REVIEW

**Comment by oleg-nenashev** [comment]
I believe it is actually a major RFE which we should highlight in changelogs and social media.
Would be great to have an explicit checklist / self0certification for https://github.com/jenkinsci/jep/blob/master/jep/2/README.adoc#selection-criteria

**Comment by casz** [comment]
Reading the selection criteria, I believe that JCasC definitely fits within the suggested plugins.

The plugin has had a sustained growth month-over-month and feedback from community suggest it is a very liked plugin.

**Review by oleg-nenashev — APPROVED** [review]
I am +1 w.r.t the request. Would appreciate feedback from @daniel-beck  and @rtyler as JEP sponsors

**Comment by MarkEWaite** [comment]
I agree that it is a major RFE that we should highlight and I agree wholeheartedly that this is a plugin that should be added.

+1 from me and I like the request for feedback from @daniel-beck and @rtyler as JEP sponsors.

**Comment by casz** [comment]
Maybe a JIRA ticket so this could be backported to the 2.204.2?

**Comment by res0nance** [comment]
+1 But I fear that with [JENKINS-51856](https://issues.jenkins-ci.org/browse/JENKINS-51856) / jenkinsci/configuration-as-code-plugin#280 still being an issue on later cores this potentially causes issues. I'm not too familiar with the issue or how often it occurs but I wanted to make sure that it was considered before this is added.

**Review by timja — APPROVED** [review]

**Comment by casz** [comment]
@res0nance we did provide a workaround in v2.34: Read here https://github.com/jenkinsci/configuration-as-code-plugin#compatibility-with-jenkins--2199

**Comment by res0nance** [comment]
@casz How often does the issue occur? It would be a terrible experience if someone was setting up jenkins the first time and it fails to load after a plugin install and reboot.

**Comment by casz** [comment]
With v2.199 it would occur every-time.

We could add additional logic that checks the Jenkins version and self introduces a easy 5-10 seconds delay as default if no value is provided.

**Comment by timja** [comment]
Why is there a need to check the version? The problem occurs on every Jenkins version it just throws an error from 2.199 onwards

**Comment by casz** [comment]
it was inside configuration as code I was thinking of improving the workaround.
Maybe we do not need the version check

**Review by alecharp — APPROVED** [review]

**Review by fcojfernandez — APPROVED** [review]
+1

**Comment by oleg-nenashev** [comment]
Do we want to proceed here? We could also meet in the middle and just do not set an "installed by default" flag while recommending it

**Comment by timja** [comment]
> Do we want to proceed here? We could also meet in the middle and just do not set an "installed by default" flag while recommending it

I can't see anyone blocking it 🤷‍♂

**Comment by casz** [comment]
@oleg-nenashev I am fine with that too :)

**Comment by varyvol** [comment]
I would go with the "middle" approach. The issue (even if somehow can be workaround) can be quite disturbing.

**Comment by MRamonLeon** [comment]
Agree with @oleg-nenashev. At least, until we get rid of the issue

**Comment by MRamonLeon** [comment]
This could be helpful to add JCasC as _suggested_: https://github.com/jenkinsci/jenkins/pull/4450

File: core/src/main/resources/jenkins/install/platform-plugins.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
         "plugins": [
             { "name": "dashboard-view" },
             { "name": "cloudbees-folder", "suggested": true },
+            { "name": "configuration-as-code", "suggested": true },
```
Comment (alecharp) [inline core/src/main/resources/jenkins/install/platform-plugins.json]:
```suggestion
            { "name": "configuration-as-code" },
```

**Review by alecharp — COMMENTED** [review]
@casz Maybe not having it to the suggested plugins until we have the Milestone merged. What do you think?

File: core/src/main/resources/jenkins/install/platform-plugins.json (hunk 1 above)
Comment (timja) [reply core/src/main/resources/jenkins/install/platform-plugins.json]: we can probably just hold a bit and see how milestones go? unless @casz wants this in now?

**Comment by casz** [comment]
As I said whatever works

File: core/src/main/resources/jenkins/install/platform-plugins.json (hunk 1 above)
Comment (timja) [reply core/src/main/resources/jenkins/install/platform-plugins.json]: Shall we make this change to add it now and then add suggested as a follow-up?

Comment (casz) [reply core/src/main/resources/jenkins/install/platform-plugins.json]: please do @timja

**Comment by casz** [comment]
Should be safe to merge :swea

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
5 approvals, and the compromise has been taken, let's merge this in,

possibly want to do a little copy-editing on the change log entry:

suggestion:
> Entry 1: Major improvement: Add configuration-as-code as a suggested plugin in the setup wizard

**Comment by daniel-beck** [comment]
> Entry 1: Major improvement: Add configuration-as-code as a suggested plugin in the setup wizard

_Suggested_ means it'll be installed by default, which is not the case in the current version (since 5 days ago). Hence I wouldn't consider it _Major_ either.

Also, once we want to make it actually _suggested_, we might want to have a conversation about it being _since 2.x_ as well, which will trigger the upgrade wizard (which nobody has seen in 200+ releases, so would require additional testing).

**Comment by oleg-nenashev** [comment]
@daniel-beck apart from next steps concerb, are you fine with merging this PR?

**Comment by daniel-beck** [comment]
Nobody cares about this list anyway, so sure 👍 

(Not that I'd be _against_ it otherwise, I just haven't seen the plugin in a while, and would probably abstain.)

**Comment by daniel-beck** [comment]
There's no relevant difference between "Suggest X Plugin" and "X Plugin is now suggested".

**Comment by oleg-nenashev** [comment]
@daniel-beck I am just trying to follow the common changelog pattern we use in https://jenkins.io/changelog/

**Comment by daniel-beck** [comment]
@oleg-nenashev I am referring to the edit I believed was made in response to https://github.com/jenkinsci/jenkins/pull/4410#issuecomment-581810877 in which I point out that in its current form, the plugin isn't _suggested_ at all.

**Pull request merged**